### PR TITLE
Cleanup GymDetail Facilities architecture

### PR DESCRIPTION
**Pull request by kevinchan159**
## Overview

Refactors view and model architecture being used for facilities

## Changes Made

- Added `FacilityType` enum
- Created `FacilityDropdown` model
- Removed `facilitiesDropdownCellStatuses` and `facilitiesDropdownCalendarSelectedIndices` which was being stored at the `GymDetailViewController` level. I moved the logic to determine status of dropdown into `FacilityDropdown` and logic of the selected calendar index into `DailyHoursRanges`
- Moved `Facility` related models into a new `Facility.swift` file
- Refactor `GymWeekCell.swift`
- Other minor code cleanup

## Test Coverage

Tested different facility sections

## Next Steps (delete if not applicable)

Make sure all facility sections actually finish

## Screenshots (delete if not applicable)

facility_refactor

### CODE REVIEW

File: Uplift/Models/Gym.swift [hunk 1]
```diff
@@ -86,8 +87,9 @@ struct Gym {
         })
 
         facilities = gymData.facilities.compactMap {
-            guard let facility = $0 else { return nil }
-            return Facility(facilityData: facility)
+            guard let facility = $0,
+                let facilityType = FacilityType(rawValue: facility.name) else { return nil }
+            return Facility(facilityData: facility, facilityType: facilityType)
```
Comment (CameronHamidi) [inline Uplift/Models/Gym.swift]: There's some repeat of code in this `init` function and the other one (specifically lines 42/61 and 73/89 for the respective functions); can we mitigate that?

File: Uplift/Views/Cells/FacilitiesDropdownCell.swift [hunk 2]
```diff
@@ -72,35 +71,36 @@ class FacilitiesDropdownCell: UICollectionViewCell {
...
             dropdownView.closeDropdown()
             headerView.rotateArrowUp()
         }
-        self.dropdownView.layoutIfNeeded()
+        dropdownView.layoutIfNeeded()
         collectionView.reloadData()
     }
 
-    static func getHeight(for facilityDetail: FacilityDetail, index: Int, calendarSelectedIndices: [Int: Int]?) -> CGFloat {
+    static func getHeight(for facilityDetail: FacilityDetail, index: Int) -> CGFloat {
```
Comment (CameronHamidi) [inline Uplift/Views/Cells/FacilitiesDropdownCell.swift]: The `index` argument is no longer needed here

File: Uplift/Models/Gym.swift (hunk 1 above)
Comment (kevinchan159) [reply Uplift/Models/Gym.swift]: theres no guarantee that `facility.name` will successfully convert to a `FacilityType` enum so I have to unwrap it outside the `init` for both

File: Uplift/Views/Cells/FacilitiesDropdownCell.swift (hunk 2 above)
Comment (kevinchan159) [reply Uplift/Views/Cells/FacilitiesDropdownCell.swift]: good catch

**Comment by kevinchan159** [comment]
> Looking good, just a few other things:
> 
> -what happened to the animations?
> 
> -there are some errors with the calendar view. One of them involves index out of bounds errors when tapping certain cells (you can test this now by tapping the Sunday for Helen Newman's pool) and the calendar disappearing entirely (see the video I posted as a response to your post in #uplift-ios)

Regarding the animations, I removed it because it looked very odd and I think doing it without animations makes it look a lot more like a dropdown.

Regarding the issues with the calendar view, that's an issue that already exists on master and is not a result of this PR.

**Pull request merged**